pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -130,24 +130,15 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pa *pav1alpha1.PodAutosc
 	pa.Status.ServiceName = sks.Status.ServiceName
 
 	// Compare the desired and observed resources to determine our situation.
-	ready, notReady := 0, 0
 	podCounter := resourceutil.NewPodAccessor(c.podsLister, pa.Namespace, pa.Labels[serving.RevisionLabelKey])
-	// Currently, SKS.IsReady==True when revision has >0 ready pods.
-	if sks.IsReady() {
-		ready, err = podCounter.ReadyCount()
-		if err != nil {
-			return fmt.Errorf("error getting ready pods %s: %w", sks.Status.PrivateServiceName, err)
-		}
-
-		notReady, err = podCounter.NotReadyCount()
-		if err != nil {
-			return fmt.Errorf("error getting not ready pods %s: %w", sks.Status.PrivateServiceName, err)
-		}
-	}
-
-	pending, terminating, err := podCounter.PendingTerminatingCount()
+	ready, notReady, pending, terminating, err := podCounter.PodCountsByState()
 	if err != nil {
-		return fmt.Errorf("error checking pods for revision %s: %w", pa.Labels[serving.RevisionLabelKey], err)
+		return fmt.Errorf("error getting pod counts %s: %w", sks.Status.PrivateServiceName, err)
+	}
+	// If SKS is not ready — ensure we're not becoming ready.
+	// TODO: see if we can perhaps propagate the SKS state to computing active status.
+	if !sks.IsReady() {
+		ready = 0
```

Comment:
I think `notReady` needs to be set to 0 as well?

Reply:
It is just for reporting, it's not used for decision making, so I am less worried about it being non 0.
And once we fix the `TODO`: ready should not be zeroed out either.